res/controllers/MVave-SMK-25-II-scripts.js

```diff
@@ -0,0 +1,277 @@
+            break;
+        }
+        case 0x5C: {
+            SMK25II.controller.activeDeck.forwardButton.inToggle();
+            break;
+        }
+        case 0x5F: {
+            SMK25II.controller.recordButton.inToggle();
+            break;
+        }
+        case 0x2E: {
+            SMK25II.controller.deckLeftButton.inToggle();
+            break;
+        }
+        case 0x2F: {
+            SMK25II.controller.deckRightButton.inToggle();
+            break;
+        }
+        case 0x4C: {
+            SMK25II.controller.activeDeck.loopButton.inToggle();
+            break;
+        }
+        default:
+            console.log(`unrecognized MMC command: ${data[4]}`);
+        }
+    };
+
+    /**
+     *
+     * @param data the full sysex message data.
+     */
+    const decodePots = function(data) {
+        switch (data[3]) {
+        case 0x60: {
+            const knob = SMK25II.controller.activeDeck.gainKnob;
+            knob.inSetParameter(knob.inValueScale(data[5]));
+            break;
+        }
+        case 0x61: {
+            const knob = SMK25II.controller.activeDeck.highKnob;
+            knob.inSetParameter(knob.inValueScale(data[5]));
+            break;
+        }
+        case 0x62: {
+            const knob = SMK25II.controller.activeDeck.midKnob;
+            knob.inSetParameter(knob.inValueScale(data[5]));
+            break;
+        }
+        case 0x63: {
+            const knob = SMK25II.controller.activeDeck.lowKnob;
+            knob.inSetParameter(knob.inValueScale(data[5]));
+            break;
+        }
+        case 0x64: {
+            const knob = SMK25II.controller.activeDeck.effectKnob;
+            knob.inSetParameter(knob.inValueScale(data[5]));
+            break;
+        }
+        case 0x65: {
+            const knob = SMK25II.controller.xfadeKnob;
+            knob.inSetParameter(knob.inValueScale(data[5]));
+            break;
+        }
+        case 0x66: {
+            const knob = SMK25II.controller.headGainKnob;
+            knob.inSetParameter(knob.inValueScale(data[5]));
+            break;
+        }
+        case 0x67: {
+            const knob = SMK25II.controller.gainKnob;
+            knob.inSetParameter(knob.inValueScale(data[5]));
+            break;
+        }
+        }
+    };
+    /**
+     *
+     * @param data the sysex message data
+     * @param length the length of the data (no longer used)
+     */
+    SMK25II.incomingData = function(data, length) {
+        if (length < 6) {
+            console.log(`expected sysex packet of length 6, got ${length}`);
+            return;
+        }
+        for (let n = 0; n < MMCHeader.length; n++) {
+            if (data[n] !== MMCHeader[n]) {
+                console.log("unknown sysex packet");
+                return;
+            }
+        }
+        if (data[length - 1] !== 0xF7) {
+            console.log("sysex packet missing trailer");
+            return;
```

Comment:
I actually wouldn't rely on length. Its a relic from when the devs thought JS was like C. Just take the `data.length` as that should always be correct.

Reply:
Fixed; thanks!